pkg/controller/certificates/util.go

```diff
@@ -118,7 +117,10 @@ func RequestMatchesSpec(req *cmapi.CertificateRequest, spec cmapi.CertificateSpe
 		violations = append(violations, "spec.ipAddresses")
 	}
 	if !util.EqualUnsorted(pki.URLsToString(x509req.URIs), spec.URIs) {
-		violations = append(violations, "spec.uriSANs")
+		violations = append(violations, "spec.uris")
+	}
+	if !util.EqualUnsorted(x509req.EmailAddresses, spec.EmailAddresses) {
+		violations = append(violations, "spec.emailAddresses")
 	}
```

Comment:
I noticed that `RequestMatchesSpec` doesn't check email addresses.
It looks like an oversight, because `PrivateKeyMatchesSpec` does check these.
Added them here without tests.
There aren't any unit tests for `RequestMatchesSpec` but I propose adding those in a followup PR.

Reply:
I think that makes sense.
But do you mean that `SecretDataAltNamesMatchSpec` checks email adresses?

Reply:
Ah, yes, you're right.